### DISCUSSION TITLE
Why doesn't expire just force the removal of the expired directory?

### DISCUSSION THREAD

**Discussion by Clara12062** [minio/minio#20553] [Q&A] [answered]
I ran into a problem: scanner had a big delay when there were a lot of files in a directory. This is the equivalent of an object that has long expired but remains uncleaned on disk.

I found that the directory has lastModified. It records the last modification time of the directory. Is it possible to scan the directory and delete it after expiration?

**Comment by klauspost** [comment, marked as answer]
No.

**Reply by Clara12062** [reply]
Is it convenient to explain why?

**Reply by klauspost** [reply]
First Dir mtime may be affected by other factors, making it unreliable. 

Second of all mtime is not part of the dirent, so you would need to open the directory inode for every entry - we do not need to do that now. 

You would need to open the metadata to confirm your decision anyway - and to do other scanning tasks, so using mtime for anything would be pointless.

We don't build software on hopes and dreams, and we don't add hacks as options, since it will inevitably lead to data loss when users that don't understand the tradeoffs use them.

**Reply by Clara12062** [reply]
Thank you very much for your reply!
I get it! Thank you very much!
